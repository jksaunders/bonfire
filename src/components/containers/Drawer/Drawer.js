import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import MaterialDrawer from "@material-ui/core/Drawer";

const CONSTANTS = {};

const propTypes = {
  children: PropTypes.node,
  // eslint-disable-next-line react/forbid-prop-types
  drawerStyles: PropTypes.object,
  onClose: PropTypes.func,
  open: PropTypes.bool.isRequired
};

const defaultProps = {
  children: null,
  drawerStyles: {},
  onClose: null
};

const StyledDrawer = styled(MaterialDrawer)`
  > .MuiDrawer-paperAnchorLeft {
    ${props => [].concat([], ...Object.keys(props.drawerStyles).map(key => `${key}: ${props.drawerStyles[key]};`))}
  }  
`;

const Drawer = ({
  children,
  onClose,
  open,
  ...rest
}) => (
  <StyledDrawer onClose={onClose || null} open={open} {...rest}>
    {children}
  </StyledDrawer>
);

Drawer.propTypes = propTypes;
Drawer.defaultProps = defaultProps;
Drawer.CONSTANTS = CONSTANTS;

export default Drawer;