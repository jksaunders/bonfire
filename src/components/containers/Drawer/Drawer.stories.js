import React, { useState } from "react";
import styled, { ThemeProvider } from "styled-components";

import { select, withKnobs } from "@storybook/addon-knobs";

import { ThemeConstants } from "../../../theming";
import {
  Button
} from "../..";
import Drawer from "./Drawer";

const StoryWrapper = styled.div``;
const StoryDrawer = styled(Drawer)`
  padding: 32px;
`;
const DrawerContent = styled.div`
  background-color: lightblue;
  color: white;
  display: flex;
  height: 100%;
  justify-content: center;
  width: 200px;
`;

const DrawerStory = () => {
  const [open, setIsOpen] = useState(false);
  return (
    <StoryWrapper>
      <Button onClick={() => setIsOpen(!open)} text="Drawer" />
      <StoryDrawer onClose={() => setIsOpen(!open)} open={open}>
        <DrawerContent>Hi!</DrawerContent>
      </StoryDrawer>
    </StoryWrapper>
  );
};

export default {
  title: "Containers|Drawer",
  component: Drawer,
  decorators: [withKnobs]
};

export const DrawerDefaultStory = () => {
  const themeMode = select("Theme Mode", Object.keys(ThemeConstants.mode.values), Object.keys(ThemeConstants.mode.values)[0]);
  const themeLayout = select("Theme Layout", Object.keys(ThemeConstants.layout.values), Object.keys(ThemeConstants.layout.values)[0]);
  const theme = { mode: themeMode, layout: themeLayout };

  return (
    <ThemeProvider theme={theme}>
      <DrawerStory />
    </ThemeProvider>
  );
};

Drawer.story = {
  name: "with text"
};