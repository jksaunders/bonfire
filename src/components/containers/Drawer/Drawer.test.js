import React from 'react';
import { expectExists, expectSnapshot, render } from '../../../testing';
import Drawer from './Drawer';

const getDrawer = (props = {}) => render(<Drawer {...props} />);

test('Drawer closed', () => {
  const { component } = getDrawer({ open: false });
  expect(component).toBe(null);
  expectSnapshot(component);
});

test('Drawer open', () => {
  const { getByRole } = getDrawer({ open: true });
  const component = getByRole('presentation');
  expectExists(component);
  expectSnapshot(component);
});
