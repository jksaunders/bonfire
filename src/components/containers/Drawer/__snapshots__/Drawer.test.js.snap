// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drawer closed 1`] = `null`;

exports[`Drawer open 1`] = `
<div
  class="MuiDrawer-root MuiDrawer-modal"
  role="presentation"
  style="position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px;"
>
  <div
    aria-hidden="true"
    class="MuiBackdrop-root"
    style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
  />
  <div
    data-test="sentinelStart"
    tabindex="0"
  />
  <div
    class="MuiPaper-root MuiDrawer-paper MuiDrawer-paperAnchorLeft MuiPaper-elevation16"
    tabindex="-1"
  />
  <div
    data-test="sentinelEnd"
    tabindex="0"
  />
</div>
`;
